### PR TITLE
Added report generation and uploading , upgraded gradle version

### PR DESCRIPTION
**Pull request by jaydee029**
For Issue galasa-dev/projectmanagement#1965
Successful PR run for this is [here](https://github.com/jaydee029/gradle/actions/runs/10471418312)

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -83,13 +83,27 @@ jobs:
...
+          name: gradle-build-log
+          path: build.log
+
+      - name: Upload JUnit Test Report
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: junit-test-report
+          path: build/reports/tests/*.html
+
```
Comment (jadecarino) [inline .github/workflows/build.yaml:106]: Lets remove this for now

**Review by jadecarino — CHANGES_REQUESTED** [review]
Please upgrade Java to 17 and remove the Jacoco step and this will be ready - thanks

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Review by jadecarino — APPROVED** [review]
Approved for building

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**